varsim.py

```diff
@@ -111,12 +111,15 @@ def makedirs(dirs):
 
 
 def monitor_multiprocesses(processes, logger):
+    is_success = True
     for p in processes:
         p.join()
         if p.exitcode != 0:
             logger.error("Process with pid %d failed with exit code %d" % (p.pid, p.exitcode))  # Marghoob: pid?
+            is_success = False
         else:
             logger.info("Process with pid %d finished successfully" % p.pid)
+    return is_success
```

Comment:
Can try throwing an exception if something fails. The exception throw must happen after all processes have finished/been killed.